### DISCUSSION TITLE
lit 3.1.0 support in the lit adapter

### DISCUSSION THREAD

**Discussion by lschierer** [withastro/roadmap#728] [Proposal]
### Body

# Summary

Lit 3.1.0 is now out, and contains breaking changes for the hydration support.  The official astro adapter will a few minimal changes to account for this.

# Background & Motivation

See 
* [the 3.0.0 changelog on github](https://github.com/lit/lit/releases/tag/lit%403.0.0)
* [the 3.1.0 changelog](https://github.com/lit/lit/releases/tag/lit%403.1.0)
* https://lit.dev/docs/v3/ssr/client-usage/
* https://lit.dev/blog/2023-10-10-lit-3.0/

# Goals

* lit users can upgrade to 3.0.0 and related packages
* Astro lit support will remain consistent with the most supported lit libraries
* hopefully fewer bugs for everyone?

**Comment by matthewp** [comment]
Please submit a PR, thank you!

**Reply by lschierer** [reply]
That seems daunting.  I'm not sure how to test such a PR before submitting it.  I've never replaced an npm package with something on my local file system in a different repository.

**Comment by lschierer** [comment]
This was closed by [#9406](https://github.com/withastro/astro/pull/9406)  Thanks for the awesome support!!!
